rmc/utils/constraint.py

```diff
@@ -955,214 +933,235 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
+    Check for significane of regional missense constraint within transcript section.
 
-    .. note::
-        Needs to be run for each break number. For example, this function needs to be run for transcripts with three breaks,
-        and it needs to be run again for transcripts with four breaks.
+    Function calculates chi square expression that assess whether observed and expected
+    missense counts in a given transcript session are significantly different than the null
+    model (no evidence of regional missense constraint).
 
-    :param ht: Input Table.
-    :param max_n_breaks: Largest number of breaks.
-    :return: Table with section and section values annotated.
-    :rtype: hl.Table
-    """
-    logger.info("Get section information for first section of each transcript...")
-    count = 1
-    section_ht = get_section_info(
-        ht,
-        section_num=count,
-        section_type="first",
-        indices=None,
-    )
+    Formula is: (section obs - section exp)^2 / section exp. Taken from ExAC RMC code.
 
-    # Check sections between breakpoint positions
-    while count <= max_n_breaks:
-        if count == 1:
-            # One break transcripts only get divided into two sections
-            # Thus, increment counter and continue
-            count += 1
-            continue
-        else:
-            temp_ht = get_section_info(
-                ht,
-                section_num=count,
-                section_type="middle",
-                indices=(count - 2, count - 1),
-            )
-            section_ht = section_ht.union(temp_ht)
-            count += 1
-    end_ht = get_section_info(ht, section_num=count, section_type="last", indices=None)
-    return section_ht.union(end_ht)
+    :param obs_expr: Total number of observed missense variants in section.
+    :param exp_expr: Total number of expected missense variants in section.
+    :return: Transcript section chi-squared value.
+    """
+    return ((obs_expr - exp_expr) ** 2) / exp_expr
 
 
-def get_unique_transcripts_per_break(
-    ht: hl.Table,
-    max_n_breaks: int,
-) -> Dict[int, Union[Set[str], hl.expr.SetExpression]]:
+def merge_rmc_hts(round_nums: List[int], is_rescue: bool) -> hl.Table:
     """
-    Return the set of transcripts unique to each break number.
+    Get table of final RMC sections in a given pathway (initial or rescue) after all break searches are complete.
 
-    If the set is empty, return an empty SetExpression.
+    :param round_nums: List of round numbers to merge results across.
+    :param is_rescue: Whether to operate on search in rescue pathway.
+    :return: Table of final RMC sections. Schema:
+        ----------------------------------------
+        Row fields:
+            'section_obs': int64
+            'section_exp': float64
+            'section_oe': float64
+            'section_chisq': float64
+            'search_type': str
```

Comment:
seeing this again made me think this likely should be a global field (will add suggestion below)